packages/ag-charts-community/src/chart/series/hierarchy/hierarchySeries.test.ts

```diff
@@ -0,0 +1,113 @@
+import type { BBox } from '../../../scene/bbox';
+import type { ChartAxisDirection } from '../../chartAxisDirection';
+import type { ChartLegendDatum, ChartLegendType } from '../../legendDatum';
+import type { SeriesNodeDataContext } from '../series';
+import type { SeriesTooltip } from '../seriesTooltip';
+import { HierarchySeries } from './hierarchySeries';
+
+class ExampleHierarchySeries extends HierarchySeries<any> {
+    override tooltip: SeriesTooltip<any> = null!;
+
+    override getSeriesDomain(_direction: ChartAxisDirection): any[] {
+        throw new Error('Method not implemented.');
+    }
+
+    override createNodeData(): Promise<SeriesNodeDataContext<any, any>[]> {
+        throw new Error('Method not implemented.');
+    }
+
+    override update(_opts: { seriesRect?: BBox | undefined }): Promise<void> {
+        throw new Error('Method not implemented.');
+    }
+
+    override getTooltipHtml(_seriesDatum: any): string {
+        throw new Error('Method not implemented.');
+    }
+
+    override getLegendData(_legendType: unknown): ChartLegendDatum<any>[] | ChartLegendDatum<ChartLegendType>[] {
+        throw new Error('Method not implemented.');
+    }
+}
+
+describe('hierarchySeries', () => {
+    it('Creates a hierarchy', async () => {
+        const series = new ExampleHierarchySeries(null!);
+        series.sizeKey = 'size';
+        series.data = [
+            { size: 5, children: [{ size: 1 }, { size: 2 }, { size: 3 }] },
+            {
+                size: 5,
+                children: [
+                    { size: 1 },
+                    { size: 2, children: [{ size: 4 }, { size: 5 }, { size: 6 }] },
+                    { size: 3, children: [{ size: 7 }] },
+                ],
+            },
+        ];
+        await series.processData();
+
+        series.rootNode.walk((node) => {
+            // @ts-expect-error
```

Comment:
Do we need this? We don't have a precedence of using these @ts ignore tags in the code-base at present so I'd like to understand if we really need these?

Reply:
Without this you get `The operand of a 'delete' operator must be optional.ts(2790)`

Reply:
I'd prefer if we did an `any` cast in that case for test purposes? (That is how we'd deal with this in other test-cases)

Reply:
Updated